### PR TITLE
Refactor Pattern to Support SPM Compilation

### PR DESCRIPTION
**Pull request by cpisciotta**
Supports both compilation commands: CompileSwift and SwiftCompile. From what I understand, the latter is now used for SPM compilation. When testing, I realized that SPM output never shows in the log.

### CODE REVIEW

File: Sources/XcbeautifyLib/Pattern.swift [hunk 1]
```diff
@@ -57,7 +57,7 @@ enum Pattern: String {
     /// $1 = file path
     /// $2 = filename (e.g. KWNull.m)
     /// $3 = target
-    case compile = #"^Compile[\w]+\s.+?\s((?:\.|[^ ])+\/((?:\.|[^ ])+\.(?:m|mm|c|cc|cpp|cxx|swift)))\s.*\((in target: (.*)|in target '(.*)' from project '.*')\)"#
+    case compile = #"^[SwiftCompile|CompileSwift].*?((?:\.|[^ ])+\/((?:\.|[^ ])+\.(?:m|mm|c|cc|cpp|cxx|swift)))\s.*\((in target: (.*)|in target '(.*)' from project '.*')\)"#
```
Comment (cpisciotta) [inline Sources/XcbeautifyLib/Pattern.swift:60]: Supports SPM compilation and a non-greedy catch-all until reaching the file path.

Comment (pepicrft) [reply Sources/XcbeautifyLib/Pattern.swift:60]: Nice improvement @cpisciotta 👏🏼

**Review by pepicrft — APPROVED** [review]

File: Sources/XcbeautifyLib/Pattern.swift (hunk 1 above)
Comment (jaysoffian) [reply Sources/XcbeautifyLib/Pattern.swift:60]: This should have been:
```
    case compile = #"^(?:SwiftCompile|CompileSwift).*?((?:\.|[^ ])+\/((?:\.|[^ ])+\.(?:m|mm|c|cc|cpp|cxx|swift)))\s.*\((in target: (.*)|in target '(.*)' from project '.*')\)"#
```

But I'm also concerned about the non-greedy operator which can cause significant performance problems due to backtracking. In fact, I'm seeing `xcbeautify` use 100% CPU now and slowing down builds significantly.

https://github.com/tuist/xcbeautify/issues/138#issuecomment-1648496774

**Comment by jaysoffian** [comment]
This project should probably have performance tests to catch regressions. I thinking by feeding it various log files and timing how long it takes to parse.

**Comment by cpisciotta** [comment]
> This project should probably have performance tests to catch regressions. I thinking by feeding it various log files and timing how long it takes to parse.

@jaysoffian Agreed. In the interim, I've reverted this change with https://github.com/tuist/xcbeautify/pull/141.

**Pull request merged**